src/Microsoft.DotNet.Interactive/Kernel.cs

```diff
@@ -529,7 +532,24 @@ internal async Task<KernelCommandResult> InvokePipelineAndCommandHandler(KernelC
             {
                 if (_commandScheduler is null)
                 {
-                    var scheduler = new KernelScheduler<KernelCommand, KernelCommandResult>();
+                    var scheduler = new KernelScheduler<KernelCommand, KernelCommandResult>(
+                        (outer, inner) =>
+                        {
+                            if (outer is null)
+                            {
+                                return false;
+                            }
+
+                            if (outer == inner.Parent)
+                            {
+                                return true;
+                            }
+
+                            var isInRoutingSlip = outer.RoutingSlip.Zip(inner.RoutingSlip, (o, i) => o.Equals(i)).All(x => x);
```

Comment:
This logic looks like a good candidate to move into `RoutingSlip`.

Reply:
Correct! Done